packages/spaces/src/helpers.js

```diff
@@ -71,3 +73,64 @@ export const updateUrl = (url, key, value) => {
 
   return `${uri}?${newParams}`;
 };
+
+/**
+ * Top Apps
+ */
+
+const TOP_APPS = {
+  ALLOWED_TYPES: ['APPLICATION', 'RESOURCE', 'NAVIGATION'],
+  BLACKLIST: ['reporting', 'how_to_guide_dental_providers', 'my_account_profile', 'my_administrators'],
+  KEYS: {
+    LAST_UPDATED: 'top-apps-updated',
+    VALUES: 'myTopApps',
+  },
+  UPDATE_EVENT: 'av:topApps:updated',
+};
+
+const getItemLocalStorage = (key) => {
+  const value = window.localStorage.getItem(key);
+  let output;
+  try {
+    output = JSON.parse(value);
+  } catch {
+    output = value;
+  }
+  return output;
+};
+
+const canTrackSpace = (spaceId, type) =>
+  TOP_APPS.ALLOWED_TYPES.some((t) => t === type) && !TOP_APPS.BLACKLIST.some((id) => id === spaceId);
+
+const getLocalStorageTopApps = (akaname) => {
+  const topAppsValues = getItemLocalStorage(`${TOP_APPS.KEYS.VALUES}-${akaname}`);
+
+  return topAppsValues;
+};
+
+export const updateTopApps = async (spaceId, type, akaname) => {
+  if (!spaceId || !type) return;
+
+  // If we can track the space
+  if (canTrackSpace(spaceId, type)) {
+    const today = dayjs();
+
+    // Grab the current top apps from localstorage
+    const topApps = (await getLocalStorageTopApps(akaname)) || {};
```

Comment:
Instead of defaulting `topApps` to `{}`, we can add a null check here since `getLocalStorageTopApps` and `getItemLocalStorage` can both return `null`. If there are no `topApps` from local storage to update, we can skip the rest of this logic instead of running it on an empty object

Reply:
Done

Reply:
Ah I missed the part where we'd actually still have a current `spaceId` to update `topApps` with, so ignore me 😅

Reply:
lol, alright. I reverted that piece but kept the refactor of getItemLocalStorage